### PR TITLE
Fix LMMAES

### PR DESCRIPTION
**Pull request by arahi10**
This pull request is for [this issue](https://github.com/RobertTLange/evosax/issues/55).

Compared with [implementation in numpy](https://github.com/NiMlr/High-Dim-ES-RL/blob/master/optimizers.py), the original LM_MA_ES in evosax seems to be incorrect wrt default parameters settings and parameters update expressions so I write a code corresponds to numpy implementation.

I used black linter and tested by evosax/tests/test_strategy_api.py. This is my first contribution, so please tell me if there are any problems.

### CODE REVIEW

File: evosax/strategies/lm_ma_es.py [hunk 1]
```diff
@@ -171,87 +151,86 @@ def tell_strategy(
         # Sort new results, extract elite, store best performer
         concat_p_f = jnp.hstack([jnp.expand_dims(fitness, 1), x])
         sorted_solutions = concat_p_f[concat_p_f[:, 0].argsort()]
+        concat_z_f = jnp.hstack([jnp.expand_dims(fitness, 1), state.z])
+        sorted_zvectors = concat_z_f[concat_z_f[:, 0].argsort()]
+        sorted_z = sorted_zvectors[:, 1:]
+        if state.best_fitness < sorted_solutions[0, 0]:
+            state.best_fitness = sorted_solutions[0, 0]
+            state.best_member = sorted_solutions[0, 1:]
```
Comment (RobertTLange) [inline evosax/strategies/lm_ma_es.py]: Can you rewrite this with `jax.lax.select` -- otherwise JAX will have to recompile for the different cases when jitting.

File: evosax/strategies/lm_ma_es.py [hunk 2]
```diff
@@ -158,7 +138,7 @@ def ask_strategy(
             state.c_d,
             state.gen_counter,
         )
-        return x, state
+        return x, state.replace(z=z)
```
Comment (RobertTLange) [inline evosax/strategies/lm_ma_es.py:140]: Maybe I am missing it, but where is the `z` needed? `(popsize, num_dims)` will require quite a lot of memory for many applications. Can we reconstruct this from the sampled `x` candidates instead?

File: evosax/strategies/lm_ma_es.py [hunk 3]
```diff
@@ -1,10 +1,10 @@
-from typing import Tuple, Optional, Union
+from typing import Optional, Tuple, Union
+
+import chex
 import jax
 import jax.numpy as jnp
-import chex
+from evosax.strategy import Strategy
```
Comment (RobertTLange) [inline evosax/strategies/lm_ma_es.py]: Can you change this back to the relative import? `from ..strategy import Strategy`

**Review by RobertTLange — COMMENTED** [review]
Thank you very much for the PR! I left some comments and recommendations.

File: evosax/strategies/lm_ma_es.py (hunk 1 above)
Comment (arahi10) [reply evosax/strategies/lm_ma_es.py]: thanks for your kindness! I rewrite this code in the latest commit (and fixed the condition in minimize problem's way).

File: evosax/strategies/lm_ma_es.py (hunk 2 above)
Comment (arahi10) [reply evosax/strategies/lm_ma_es.py:140]: Yes, this modification is the largest difference between original evosax's code and numpy code for memory size.
We have to use ```z``` to update ```p_sigma,M```(pls see the update functions), but in order to get ```z``` from solution vectors we have to calculate inverse matrices and that may causes both computing errors and cost. I think the procedure for getting z from solutions will be like this;
  iteratively calculate inverse of``` (1-c_d[j])*I+c_d[j]*jnp.outer(M[j,:],M[j,:])``` like ```sample``` function and select j by checking   generation.


Are there any linear algebra tricks to keep away from calculation of inverse matrices? I can't came up with that.

File: evosax/strategies/lm_ma_es.py (hunk 3 above)
Comment (arahi10) [reply evosax/strategies/lm_ma_es.py]: thanks for your reviewing! I changed it in the latest commit.

File: evosax/strategies/lm_ma_es.py (hunk 1 above)
Comment (arahi10) [reply evosax/strategies/lm_ma_es.py]: Sorry, I found that ```Strategy.tell``` function updates them properly so these code are unnecessary; in the latest commit it is removed.

File: evosax/strategies/lm_ma_es.py (hunk 2 above)
Comment (RobertTLange) [reply evosax/strategies/lm_ma_es.py:140]: Mhmm, I am not sure -- but I guess this is a bit of a problem. I guess it will be hard to scale to neuroevolution even though it is the limited memory version :( I will give this more of a thought.

Comment (arahi10) [reply evosax/strategies/lm_ma_es.py:140]: Hello. If possible, could you please approve the workflow and run the test before devising the implementation? I am wondering if there are any implementation mistakes at this time.
